### ISSUE TITLE
XBee 900HP

### ISSUE THREAD

**Issue by ajgriesemer** [jefffhaynes/XBee#6]
Hi Jeff,

I'm having issues getting the library to work with an XBee 900HP. When I plug in a Series 1 everything works great, but my application requires the 900HP.

The 900HP device is discovered correctly, but as soon as a packet of data comes in I get an exception. I have the endpoint device configured for cyclic sampling so my packets are coming in as RxIndicatorSampleExtFrame. 

I noticed that in the source for RxIndicatorSampleExtFrame.cs the valuePath for the SerializeWhen attribute on the AnalogSamples is "AnalogSampleChannels" when the name of the field is AnalogChannels (AnalogSampleChannels is the field type). I changed "AnalogSampleChannels" to "AnalogChannels", but I still get an exception in GetAnalogSamples:
"Value cannot be null.Parameter name: first"

I've been trying to track down what else might be wrong by working my way through the Binary Serializer, but haven't been able to find much.

How much have you tested this against the 900HP series? Would you expect to see these kinds of issues?

**Comment by jefffhaynes** [comment]
I would say I've tested it "somewhat" :)
I'm just getting on a plane and I'll be away from hardware for a few days but I'll try to look at it as soon as possible. It does sound like there's a bug although the exception you got after the change doesn't sound familiar. Kind of sounds like a linq exception. 

If you figure out anything else in the meantime let me know. I also wrote the serializer so it's possible it's a problem there (although that's much better tested). Make sure you're updated to the 4.6.4 serializer.
Thanks,
Jeff
    _____________________________
